Modules/ThirdParty/Eigen3/src/itkeigen/CMakeLists.txt

```diff
@@ -758,6 +758,12 @@ if(ITK_USE_EIGEN_MPL2_ONLY)
   target_compile_definitions (eigen_internal INTERFACE "EIGEN_MPL2_ONLY")
 endif()
 
+# Hack to disable warning. To remove when/if fixed upstream.
+# See issue: https://gitlab.com/libeigen/eigen/-/issues/2416
+if(CMAKE_CXX_STANDARD GREATER_EQUAL 17 AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  target_compile_definitions (eigen_internal INTERFACE "-Wnoused-but-marked-unused")
```

Comment:
shouldn't that have another dash: `-Wno-used-but-marked-unused`?

Reply:
Also, will this cause 'unknown warning' warnings on older clangs?

Reply:
> shouldn't that have another dash: -Wno-used-but-marked-unused?

Indeed, fixed.

> Also, will this cause 'unknown warning' warnings on older clangs?

The attribute was introduced with c++17, so I am guessing any older clang will fail to compile with CXX_STANDARD < 17 anyway.

Reply:
Yeah, probably. If not, hopefully we'll have a fix from upstream by then.